### PR TITLE
Solutions for Chapter 1

### PR DESCRIPTION
**Pull request by keyridan**
### Solutions for Chapter 1
Thank you for such a great course!
cc @vrom911 @chshersh

### CODE REVIEW

File: src/Chapter1.hs [hunk 1]
```diff
@@ -71,7 +71,7 @@ Modules should have the same name as the corresponding file with
 the `.hs` extension.
 -}
 module Chapter1 where
-
+import Data.List
```
Comment (vrom911) [inline src/Chapter1.hs:74]: Intriguing! 👀

File: src/Chapter1.hs [hunk 2]
```diff
@@ -541,8 +542,14 @@ value after "=" where the condition is true.
...
 -}
+mid :: Int -> Int -> Int -> Int
+mid x y z
+    | x <= y && y <= z || x > y && y > z = y
+    | y <= x && x <= z || y > x && x > z = x
+    | otherwise = z
 
-mid x y z = error "mid: not implemented!"
+midWithSort :: Int -> Int -> Int -> Int
+midWithSort x y z = sort [x,y,z] !! 1
```
Comment (vrom911) [inline src/Chapter1.hs:552]: Oh I see why `Data.List` 😄 
Very creative solution 🧠 🎨

File: src/Chapter1.hs [hunk 3]
```diff
@@ -556,8 +563,13 @@ True
...
 False
 -}
-isVowel c = error "isVowel: not implemented!"
+isVowel :: Char -> Bool
+isVowel c
+    |c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' = True
+    |otherwise = False
 
+isVowelWithoutGuards :: Char -> Bool
+isVowelWithoutGuards c = c `elem` "aeiou"
```
Comment (vrom911) [inline src/Chapter1.hs:572]: Yeah, both of these solutions are valid! Nice catch 😉 
One note in here, that sometimes, `elem` could be slower than the explicit pattern matching. I remember there were some benchmarks on one particular case, that showed how moving to pattern matching on each case separately drastically decrease time 🐎

File: src/Chapter1.hs [hunk 4]
```diff
@@ -619,9 +631,11 @@ Implement a function that returns the sum of the last two digits of a number.
 Try to introduce variables in this task (either with let-in or where) to avoid
 specifying complex expressions.
 -}
-
-sumLast2 n = error "sumLast2: Not implemented!"
-
+sumLast2 :: Int -> Int
+sumLast2 n =
+    let (withoutLast, last) = abs n `divMod` 10
+        secondToLast = withoutLast `mod` 10
```
Comment (vrom911) [inline src/Chapter1.hs:637]: Great usage of standard functions 👌🏼 

You can notice that actually second last could be written with the `lastDigit` functions from the previous task. But as you correctly noticed, `abs` id not required in here, so this is already a great solution 👍🏼

File: src/Chapter1.hs [hunk 5]
```diff
@@ -640,9 +654,9 @@ Implement a function that returns the first digit of a given number.
 You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
-
-firstDigit n = error "firstDigit: Not implemented!"
-
+firstDigit :: Int -> Int
+firstDigit n = if value < 10 then value else firstDigit (value `div` 10)
+  where value = abs n
```
Comment (vrom911) [inline src/Chapter1.hs:659]: Great use of `where` construction 💯

**Review by vrom911 — APPROVED** [review]
Amazing job! Congrats with finishing Chapter One 🥳 
Very clean and thoughtful solutions, bravo 👏🏼

File: src/Chapter1.hs (hunk 3 above)
Comment (keyridan) [reply src/Chapter1.hs:572]: Wow, good to know, thanks!

**Pull request merged**